var/spack/repos/builtin/packages/qgis/package.py

```diff
@@ -111,6 +113,7 @@ class Qgis(CMakePackage):
 
     # Take care of conflicts using depends_on
     depends_on('proj@5:', when='@3.8.2:')
+    depends_on('proj@6.3.1:', when='@3.12.1:')    # Not sure of the when= version.
```

Comment:
@Sinan81 might know where to find the exact version requirements.

Reply:
For dependencies I check out either of the following documents which are retrieved from branch `final-3_12_1` for example:

- https://github.com/qgis/QGIS/blob/final-3_12_1/INSTALL
- http://htmlpreview.github.io/?https://raw.github.com/qgis/QGIS/final-3_12_1/doc/INSTALL.html

every release has a corresponding `final-...` tag:
```
$ git clone https://github.com/qgis/QGIS.git
$ cd QGIS
$ git tag
...
final-3_12_2
final-3_2_0
final-3_2_1
final-3_2_2
final-3_2_3
final-3_4_0
final-3_4_1
final-3_4_10
final-3_4_11
final-3_4_12
final-3_4_13
final-3_4_14
final-3_4_15
final-3_4_2
final-3_4_3
final-3_4_4
final-3_4_5
final-3_4_6
final-3_4_7
final-3_4_8
final-3_4_9
final-3_6_0
final-3_6_1
final-3_6_2
final-3_6_3
final-3_8_0
final-3_8_1
final-3_8_2
final-3_8_3
...
```